python/delta/tables.py

```diff
@@ -581,6 +581,14 @@ def restoreToTimestamp(self, timestamp: str) -> DataFrame:
             getattr(self._spark, "_wrapped", self._spark)  # type: ignore[attr-defined]
         )
 
+    @since(1.3)  # type: ignore[arg-type]
+    def optimize(self) -> "DeltaOptimizeBuilder":
+        """
+        Optimize the data layout of the table.
```

Comment:
You can add more details here similar to Scala API: https://github.com/delta-io/delta/blob/master/core/src/main/scala/io/delta/tables/DeltaTable.scala#L186

Reply:
Added more details to the docstrings. I also rebased to the master.